src/python/WMComponent/WorkflowUpdater/WorkflowUpdaterPoller.py

```diff
@@ -26,6 +31,161 @@
+    :param fname: name of the file to extract
+    :return: content of file
+    """
+    mode = tarMode(tfile, 'r')
+    with tarfile.open(tfile, mode, encoding='utf-8') as tar:
+        f = tar.extractfile(fname)
+        data = f.read()
+        # convert our data bytes into JSON object
+        return json.loads(data)
+
+
+def blockLocations(jdoc):
+    """
+    Return dict block names and their location from provided json sandbox file.
+
+    json structure is
+    {"<type: mc>": {"<blockName>": {"FileList" [], "NumberOfEvents":1, "PhEDExNodeNames": []},
+                    "<blockName>": {"FileList" [], "NumberOfEvents":1, "PhEDExNodeNames": []}, ...}
+
+    :param jdoc: JSON document
+    :return: dict {'blockName': [rses], ...}
+    """
+    bdict = {}
+    for rec in jdoc.values():
+        for key in rec.keys():
+            doc = rec[key]
+            bdict[key] = doc['PhEDExNodeNames']
+    return bdict
+
+
+def rucioBlockLocations(rucio, blocks):
+    """
+    Return dictionary in form: {"path-to-json1": jdoc1, "path-to-json2": jdoc2}
+    :param rucio: rucio client
+    :param blocks: list of blocks
+    :return: directionary
+    """
+    # rucio returns: [{"name": blockName, "replica": list(set(rses))}, ...]
+    rucioList = rucio.getReplicaInfoForBlocks(block=blocks)
+    bdict = {}
+    for item in rucioList:
+        bdict[item['name']] = item['replica']
+    return bdict
+
+
+def checkChanges(rdict, bdict):
+    """
+    Compare if provided rucio block locations are different from block locations presented
+    in pileup configuration JSON file.
+    :param rdict: dict of blocks locations from rucio
+    :param bdict: dict of blocks locations from pileupconf.json sandbox file
+    :return: boolean (i.e. if there are changes or not)
+    """
+    # compare block names
+    if len(rdict) != len(bdict):
+        return True
+
+    # compare locations
+    for key in rdict.keys():
+        if sorted(rdict[key]) != sorted(bdict[key]):
+            return True
+    return False
+
+
+def updateBlockInfo(jdoc, rdict, bdict):
+    """
+    Discard given pileup names from JSON sandbox and return new dict.
+    NOTE: this function may consume lots of memory due to jdoc structure which
+    we need to parse and create new dict since we can't discard keys in place 
+    of jdoc internal dictionary, i.e.
+
+    jdoc structure is
+    {"<type: mc>": {"<block>": {"FileList" [], "NumberOfEvents":1, "PhEDExNodeNames": []},
+                    "<block>": {"FileList" [], "NumberOfEvents":1, "PhEDExNodeNames": []}, ...}
+    and we need to discard block names. They are located in internal dictionary
+    and in place pop-up of jdoc is not allowed since it will change size of internal dictionary.
+    Therefore, to overcome this obstacles we need to copy relevant pileupName key:values
+    into new dict structure and return new dict. Such operation will mostly double the
+    memory size.
+
+    :param jdoc: JSON sandbox dictionary
```

Comment:
I feel like I got lost with `jdoc` and `bdict`. Aren't they supposed to be the same thing? Json from the sandbox and pileupconf.json is the same thing.

As already expressed in other parts of this PR, the variable names that you pick don't make it easier to understand the code. Please consider refactoring them and and for future PRs.

Reply:
In my implementation I used both `json` and blocks from it which I called `bdict`. Even though we agreed that mine implementation and yours are the same in terms, I see that in your implementation I don't need `bdict`, therefore it is beneficial and I switched now to it. And, it means that this comment is resolved too.